nukebuild/Build.cs

```diff
@@ -0,0 +1,278 @@
+        .DependsOn(Compile)
+        .Executes(() =>
+        {
+
+            var dotMemoryUnitPath =
+                ToolPathResolver.GetPackageExecutable("JetBrains.dotMemoryUnit", "dotMemoryUnit.exe");
+            var xunitRunnerPath =
+                ToolPathResolver.GetPackageExecutable("xunit.runner.console", "xunit.console.x86.exe");
+            var args = new[]
+            {
+                Path.GetFullPath(xunitRunnerPath),
+                "--propagate-exit-code",
+                "--",
+                "tests\\Avalonia.LeakTests\\bin\\Release\\net461\\Avalonia.LeakTests.dll"
+            };
+            var cargs = string.Join(" ", args.Select(a => '"' + a + '"'));
+
+            var proc = Process.Start(new ProcessStartInfo(dotMemoryUnitPath, cargs)
+            {
+                UseShellExecute = false
+            });
+
+            if (!proc.WaitForExit(120000))
+            {
+                proc.Kill();
+                throw new Exception("Leak tests timed out");
+            }
+
+            var leakTestsExitCode = proc.ExitCode;
+            
+            if (leakTestsExitCode != 0)
+            {
+                throw new Exception("Leak Tests failed");
+            }
+            
+
+            var testAssembly = "tests\\Avalonia.LeakTests\\bin\\Release\\net461\\Avalonia.LeakTests.dll";
+            DotMemoryUnit(
+                $"{XunitPath.DoubleQuoteIfNeeded()} --propagate-exit-code -- {testAssembly}",
+                timeout: 120_000);
+        });
+
+    Target ZipFiles => _ => _
+        .After(CreateNugetPackages, Compile, RunCoreLibsTests, Package)    
+        .Executes(() =>
+        {
+            var data = Parameters;
+            Zip(data.ZipCoreArtifacts, data.BinRoot);
+
+            Zip(data.ZipNuGetArtifacts, data.NugetRoot);
+
+            Zip(data.ZipTargetControlCatalogDesktopDir,
+                GlobFiles(data.ZipSourceControlCatalogDesktopDir, "*.dll").Concat(
+                    GlobFiles(data.ZipSourceControlCatalogDesktopDir, "*.config")).Concat(
+                    GlobFiles(data.ZipSourceControlCatalogDesktopDir, "*.so")).Concat(
+                    GlobFiles(data.ZipSourceControlCatalogDesktopDir, "*.dylib")).Concat(
+                    GlobFiles(data.ZipSourceControlCatalogDesktopDir, "*.exe")));
+        });
+
+    Target CreateNugetPackages => _ => _
+        .DependsOn(Compile)
+        .After(RunTests)
+        .Executes(() =>
+        {
+            if (Parameters.IsRunningOnWindows)
+
+                MSBuild(Parameters.MSBuildSolution, c => c
+                    .SetConfiguration(Parameters.Configuration)
+                    .SetVerbosity(MSBuildVerbosity.Minimal)
+                    .AddProperty("PackageVersion", Parameters.Version)
+                    .AddProperty("iOSRoslynPathHackRequired", "true")
+                    .SetToolsVersion(MSBuildToolsVersion._15_0)
+                    .AddTargets("Restore", "Pack"));
```

Comment:
We shouldn't need the restore here. If we do, then the same situation as mentioned above with the build target also applies.